server/index.js

```diff
@@ -150,6 +150,30 @@ app.post('/api/add-entry', (req, res, next) => {
     .catch(err => next(err));
 });
 
+/**
+ * route returns a list of total sum spent for each month by the user
+ */
+app.get('/api/monthly-expense/:userId', (req, res, next) => {
+
+  const sql = `select sum("amount"), TO_CHAR("date" :: DATE, 'Monthyyyy') as "date"
+               from (select "amount", "date" from "entry" where "userId" = $1) as "userEntries"
+               group by TO_CHAR("date" :: DATE, 'Monthyyyy')`;
```

Comment:
I could be wrong, but i think you can simplify this to skip the subquery.
```sql
select sum("amount") as "sum",
       to_char("date"::date, 'Monthyyyy') as "month" -- this column should probably be stored as a date so you don't have to cast it with the ::
  from "entry"
 group by "month"; -- maybe you need to put the to_char here too?
```

Reply:
I used the subquery to get all the entry that belongs to a specific user. If I do a group without it, it will include all entries in the entry table.